ds3-autogen-utils/src/main/java/com/spectralogic/ds3autogen/utils/MultiResponseSplitConverter.java

```diff
@@ -0,0 +1,529 @@
+ *   Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ *   this file except in compliance with the License. A copy of the License is located at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file.
+ *   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ *   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ *   specific language governing permissions and limitations under the License.
+ * ****************************************************************************
+ */
+
+package com.spectralogic.ds3autogen.utils;
+
+import com.google.common.collect.ImmutableList;
+import com.spectralogic.ds3autogen.api.models.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.spectralogic.ds3autogen.utils.ConverterUtil.hasContent;
+import static com.spectralogic.ds3autogen.utils.ConverterUtil.isEmpty;
+import static com.spectralogic.ds3autogen.utils.Helper.stripPath;
+
+/**
+ * Converts Ds3Requests with multiple response types for a given response code
+ * into separate Ds3Requests, each with only one response type.
+ */
+public final class MultiResponseSplitConverter {
+
+    private static final Logger LOG = LoggerFactory.getLogger(MultiResponseSplitConverter.class);
+
+    public static final String NAMESPACE_FULL_DETAILS = "FullDetails";
+
+    //Names of known request handlers that have two response types for a given response code
+    private static final String GET_PHYSICAL_PLACEMENT_FOR_OBJECTS = "GetPhysicalPlacementForObjectsRequestHandler";
+    private static final String VERIFY_PHYSICAL_PLACEMENT_FOR_OBJECTS = "VerifyPhysicalPlacementForObjectsRequestHandler";
+    private static final String GET_TAPE_PARTITION = "GetTapePartitionRequestHandler";
+    private static final String GET_TAPE = "GetTapeRequestHandler";
+
+    private MultiResponseSplitConverter() { }
+
+    /**
+     * Splits Ds3Requests with multiple response types for a given response code into
+     * distinct Ds3Requests with only one response type for each response code.
+     * @param ds3ApiSpec A spec containing a list of Ds3ApiRequests to convert
+     */
+    public static Ds3ApiSpec splitAllMultiResponseRequests(final Ds3ApiSpec ds3ApiSpec) {
+        return new Ds3ApiSpec(
+                splitAllMultiResponseRequests(ds3ApiSpec.getRequests()),
+                ds3ApiSpec.getTypes());
+    }
+
+    /**
+     * Splits Ds3Requests with multiple response types for a given response code into
+     * distinct Ds3Requests with only one response type for each response code.
+     * @param ds3Requests A list of Ds3Requests to be converted
+     */
+    public static ImmutableList<Ds3Request> splitAllMultiResponseRequests(
+            final ImmutableList<Ds3Request> ds3Requests) {
+        if (isEmpty(ds3Requests)) {
+            LOG.info("No requests");
```

Comment:
Change this to say "No requests to split" to differentiate the error message from some of the others.